### PR TITLE
Add posts news

### PR DESCRIPTION
**Pull request by koechkevin**
## Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Screenshots
image

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation

### CODE REVIEW

**Review by kilemensi — COMMENTED** [review]
👍🏽 

---

Quick review from me: for these posts, a user need to enter long form **_content_** similar to Ghost or WordPress, not just **_description_**. At least, we should have rich text and media.

As always, [examples on Payload repos](https://github.com/payloadcms/website-cms/blob/main/src/collections/Posts.ts) are always a good place to start.

File: apps/charterafrica/src/components/PostCard/PostCard.js [hunk 1]
```diff
@@ -44,10 +44,12 @@ const PostCard = React.forwardRef(function PostCard(props, ref) {
             variant="h5SmallSemiBold"
             sx={(theme) => ({
               mb: 2.5,
+              fontWeight: 400,
               minHeight: `calc(${theme.typography.h5SmallSemiBold.fontSize}px * ${theme.typography.h5SmallSemiBold.lineHeight} * 3)`,
               [theme.breakpoints.up("md")]: {
                 minHeight: `calc(${theme.typography.h5SemiBold.fontSize}px * ${theme.typography.h5SemiBold.lineHeight} * 3)`,
                 typography: "h5SemiBold",
+                fontWeight: 400,
```
Comment (kilemensi) [inline apps/charterafrica/src/components/PostCard/PostCard.js:52]: Confirm with the designer if there is an issue here: Semibold is 600. If they want 400, that's Regular
[
Common font weight name mapping](https://developer.mozilla.org/en-US/docs/Web/CSS/font-weight#common_weight_name_mapping)

**Comment by kilemensi** [comment]
Why are you asking for reviews when the PR is still under draft @koechkevin.

But a quick review:

1. How do we mark a news article as featured post? I'm thinking there should be block on news page in the CMS where content editors can select a news article(s) to feature but your move.
2. News collection must be readable to anyone, otherwise we may end up with auth issues.
3. Do we need anything more than name on author(s)? Can a post have more than one author? Will have Authors as a separate collection help? I'm thinking of having to retype an author every time may lead to spelling mistakes, etc.

**Comment by koechkevin** [comment]
> Why are you asking for reviews when the PR is still under draft @koechkevin.
> 
> But a quick review:
> 
> 1. How do we mark a news article as featured post? I'm thinking there should be block on news page in the CMS where content editors can select a news article(s) to feature but your move.
> 2. News collection must be readable to anyone, otherwise we may end up with auth issues.
> 3. Do we need anything more than name on author(s)? Can a post have more than one author? Will have Authors as a separate collection help? I'm thinking of having to retype an author every time may lead to spelling mistakes, etc.

I clicked re review by mistake and no way to revert it so I marked it as draft.

1. Yes, a featured post will be a block.
2. 👍
3. Having authors as a separate collection can help.

**Comment by kilemensi** [comment]
> I clicked re review by mistake and no way to revert it so I marked it as draft.
> 

Click the Reviewers gear icon and deselect the selected reviewers @koechkevin 
Screenshot 2023-02-22 at 12 30 16

File: apps/charterafrica/src/payload/fields/post.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    label: {
+      en: "Image",
+      pt: "Imagem",
+      fr: "Image",
+    },
+    type: "upload",
+    relationTo: "media",
+    required: true,
+  },
+  richText({
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/fields/post.js:40]: We need the full `content` here in addition to excerpt/summary. The full content should probably be a block that allows various fields such as richText, media and video/ifram embed.

File: apps/charterafrica/src/payload/fields/post.js [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+
+const postFields = [
+  {
+    name: "title",
+    label: {
+      en: "Title",
+      fr: "Titre",
+      pt: "Título",
+    },
+    type: "textarea",
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/fields/post.js:17]:
```suggestion
    type: "text",
```

Title doesn't need to be that long

File: apps/charterafrica/src/payload/fields/post.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  slug(),
+  {
+    name: "authors",
+    required: true,
+    type: "relationship",
+    relationTo: "author",
+    hasMany: true,
+  },
+  {
+    name: "image",
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/fields/post.js:30]: This should probably be featured image or cover image.

File: apps/charterafrica/src/payload/collections/News.js [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import fields from "../fields/post";
+
+const News = {
+  slug: "news",
+  admin: {
+    useAsTitle: "title",
+    defaultColumns: ["title", "author"],
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/collections/News.js:7]: Lets add a publication date here as well so that content editor can sort most recent posts, etc.

File: apps/charterafrica/src/payload/collections/Authors.js [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  slug: "author",
+  access: {
+    read: () => true,
+  },
+  admin: {
+    useAsTitle: "name",
+  },
+  fields: [
+    {
+      name: "name",
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/collections/Authors.js:11]: If we're going for just name, then lets make it full name at least.

File: apps/charterafrica/src/payload/blocks/FeaturedNewsPost.js [hunk 7]
```diff
@@ -0,0 +1,30 @@
+const FeaturedNewsPost = {
+  slug: "featured-news-post",
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/blocks/FeaturedNewsPost.js:2]: If relationship field allows relating to more than one collection, we can call this just FeaturedPost and we'll reuse it for News, Research and Events collections

File: apps/charterafrica/src/pages/[...slugs].page.js [hunk 8]
```diff
@@ -22,60 +22,40 @@ import Resources from "@/charterafrica/components/Resources";
 import Spotlight from "@/charterafrica/components/Spotlight";
 import { getPageServerSideProps } from "@/charterafrica/lib/data";
 
+const componentMap = {
```
Comment (kilemensi) [inline apps/charterafrica/src/pages/[...slugs].page.js:25]: componentsBySlugs

File: apps/charterafrica/src/lib/data/common/index.js [hunk 9]
```diff
@@ -171,48 +171,34 @@ export async function processPageFellowships({ blocks }) {
...
-    link: {
-      href: "/knowledge/news",
-    },
+export async function processPageNews({ blocks }, api) {
+  const { docs } = await api.getCollection("news");
+
+  const processArticle = (data) => ({
+    ...data,
+    author: data?.authors?.map(({ name }) => name).join(", ") ?? null,
+    date: new Date(data?.publishedOn || data?.createdAt).toUTCString(),
```
Comment (kilemensi) [inline apps/charterafrica/src/lib/data/common/index.js:180]: If publishedOn is required, why do we need to do this?

**Review by kilemensi — COMMENTED** [review]
👍🏽 @koechkevin

File: apps/charterafrica/src/payload/fields/post.js [hunk 10]
```diff
@@ -0,0 +1,65 @@
...
+  richText({
+    name: "excerpt",
+    label: {
+      en: "Excerpt",
+      fr: "Extrait",
+      pt: "Excerto",
+    },
+    localized: true,
+  }),
+  linkGroup(),
```
[the remark this answers is not in the document]
Comment (koechkevin) [reply apps/charterafrica/src/payload/fields/post.js:49]: Intended to link to view post content e.g `/post/{postId}` @kilemensi

[the remark this answers is not in the document]
Comment (kilemensi) [reply apps/charterafrica/src/payload/fields/post.js:49]: Not sure I follow @koechkevin : A link to _view_ a **post** is going to be _manually_ entered as a field in a **Post**?

[the remark this answers is not in the document]
Comment (koechkevin) [reply apps/charterafrica/src/payload/fields/post.js:49]: Actually, you are right we do not need it.

File: apps/charterafrica/src/payload/fields/post.js (hunk 2 above)
Comment (koechkevin) [reply apps/charterafrica/src/payload/fields/post.js:40]: @kilemensi won't video embed be the same as media upload?

Comment (kilemensi) [reply apps/charterafrica/src/payload/fields/post.js:40]: No @koechkevin ... video embeds usually use youtube or similar services. People shouldn't be uploading videos to the CMS.

Comment (koechkevin) [reply apps/charterafrica/src/payload/fields/post.js:45]: So we only give users a text field to enter embed URL. Right?

Comment (kilemensi) [reply apps/charterafrica/src/payload/fields/post.js:45]: Yeah, what does Ghost/WP give? Anyways, may be it should be a code field similar to the newsletter?

Comment (koechkevin) [reply apps/charterafrica/src/payload/fields/post.js:46]: We can give users to select the type of embed they'd like to use.

Either pasting custom url or embed code. The approach we used on links

**Review by kelvinkipruto — APPROVED** [review]
LGTM!!🚀

File: apps/charterafrica/src/lib/data/common/index.js [hunk 11]
```diff
@@ -171,48 +171,41 @@ export async function processPageFellowships({ blocks }) {
...
+  const featuredNewsPost = processArticle(
+    blocks.find(({ slug }) => slug === "featured-post")?.featuredPost?.value ??
+      null
+  );
+
+  const featuredPost = {
+    category: "News",
+    ...featuredNewsPost,
+    slug: "featured-post",
+  };
```
Comment (kilemensi) [inline apps/charterafrica/src/lib/data/common/index.js:201]: Why don't we check if we have data first before processing it? This code looks like it will create an empty `featuredPost` (with category and slug) even if the original page data didn't have `featured-post` block.

File: apps/charterafrica/src/lib/data/common/index.js [hunk 12]
```diff
@@ -349,8 +342,9 @@ export async function getPageProps(context, api) {
       })
     )) || [];
   const processPage = processPageFunctionsMap[page.slug];
+  let processedPage = {};
   if (processPage) {
-    await processPage(page, api);
+    processedPage = await processPage(page, api);
```
Comment (kilemensi) [inline apps/charterafrica/src/lib/data/common/index.js:347]: `processPage` can & should modify the page or why do we need to create a copy?

File: apps/charterafrica/src/payload/blocks/BlogContent.js [hunk 13]
```diff
@@ -0,0 +1,14 @@
+import blockFields from "../fields/blockFields";
+import richText from "../fields/richText";
+
+const BlogContent = {
+  slug: "blogContent",
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/blocks/BlogContent.js:5]: I wouldn't call this BlogContent since we'll use it Reseach, privacy policy page, etc. etc. May be RichText, etc.

File: apps/charterafrica/src/payload/blocks/BlogContent.js [hunk 14]
```diff
@@ -0,0 +1,14 @@
...
+  fields: [
+    blockFields({
+      name: "blogContentFields",
+      fields: [richText({ name: "content", admin: { elements: ["ul"] } })],
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/blocks/BlogContent.js:9]: This richText should include pretty much everything, headers, links, etc. Not sure about images if we have a separate block for that.

File: apps/charterafrica/src/payload/blocks/FeaturedPost.js [hunk 15]
```diff
@@ -0,0 +1,30 @@
...
+    {
+      name: "featuredPost",
+      label: {
+        en: "Featured Post",
+        pt: "Postagem em destaque",
+        fr: "Après sélectionnée",
+      },
+      type: "relationship",
+      relationTo: ["news"],
+      required: true,
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/blocks/FeaturedPost.js:25]: Have you checked if we can have multiple `relationTo` and the data structure wouldn't change when we add those? Remember, once data is entered by the content team, it will be very difficult to change its structure afterwards.

File: apps/charterafrica/src/payload/blocks/VideoEmbed.js [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+    blockFields({
+      name: "videoEmbedField",
+      fields: [
+        {
+          type: "row",
+          fields: [
+            {
+              name: "embedType",
+              type: "radio",
+              defaultValue: "url",
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/blocks/VideoEmbed.js:15]: Which services do you know only share url? I just want to be sure if this is something we should be supporting now or in the future.

File: apps/charterafrica/src/payload/blocks/VideoEmbed.js [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+            },
+          ],
+        },
+        {
+          name: "url",
+          label: {
+            en: "Video URL",
+            fr: "URL de la vidéo",
+            pt: "URL do vídeo",
+          },
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/blocks/VideoEmbed.js:43]: Thinking out loud, would embedding video, audio or even a chart be any different on the front-end?

File: apps/charterafrica/src/payload/fields/post.js [hunk 18]
```diff
@@ -0,0 +1,76 @@
...
+    required: true,
+  },
+  slug(),
+  {
+    name: "authors",
+    required: true,
+    type: "relationship",
+    relationTo: "author",
+    hasMany: true,
+  },
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/fields/post.js:31]: Move this to the bottom or to the side panel

File: apps/charterafrica/src/payload/fields/post.js [hunk 19]
```diff
@@ -0,0 +1,76 @@
...
+    },
+    type: "upload",
+    relationTo: "media",
+    required: true,
+    filterOptions: {
+      mimeType: { contains: "image" },
+    },
+  },
+  richText({
+    name: "excerpt",
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/fields/post.js:47]: 👇🏽 is how Excerpt look right now.
Screenshot 2023-02-24 at 13-26-24 Creating - News - Payload

That's how full content should look like. excerpt is like summary; shouldn't contain heading, images or other uploads, etc. Probably should just contain `leaves` i.e. bold, italic, etc.

File: apps/charterafrica/src/payload/fields/post.js [hunk 20]
```diff
@@ -0,0 +1,76 @@
...
+    localized: true,
+  }),
+  {
+    name: "content",
+    type: "blocks",
+    blocks: [BlogContent, MediaBlock, VideoEmbed],
+    required: true,
+  },
+  {
+    name: "publishedOn",
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/fields/post.js:62]: This should be required.

**Review by kilemensi — COMMENTED** [review]
👍🏽 

---

Show we support uploads/media in the `richText` field for content? I mean we have a separate block for that. My main worry is can we really control alignment and other visual aspects in an image is mixed up with other text content?

File: apps/charterafrica/src/payload/blocks/VideoEmbed.js (hunk 17 above)
Comment (koechkevin) [reply apps/charterafrica/src/payload/blocks/VideoEmbed.js:43]: I think they'd follow a similar approach @kilemensi

File: apps/charterafrica/src/payload/blocks/VideoEmbed.js (hunk 16 above)
Comment (koechkevin) [reply apps/charterafrica/src/payload/blocks/VideoEmbed.js:15]: I saw from wordpress pasting a youtube url works for them. Apparently they use [oEmbed Protocol](https://oembed.com/)

File: apps/charterafrica/src/payload/blocks/FeaturedPost.js (hunk 15 above)
Comment (koechkevin) [reply apps/charterafrica/src/payload/blocks/FeaturedPost.js:25]: @kilemensi Here, [docs](https://payloadcms.com/docs/fields/relationship) explain how data is saved for dynamic references.

Comment (kilemensi) [reply apps/charterafrica/src/payload/blocks/FeaturedPost.js:25]: I know about the docs @koechkevin ... I wanted **you** to go through it and arrive at either  **Has One - Polymorphic** if we're only ever going to have one featured post or **Has Many - Polymorphic** if we're allowing multiple featured docs

I think we're missing `hasMany`.

File: apps/charterafrica/src/lib/data/common/index.js (hunk 12 above)
Comment (koechkevin) [reply apps/charterafrica/src/lib/data/common/index.js:352]: The other workaround if we need to mutate page object directly is do something like

```js
  blocks[0] = featuredPost;
  blocks[1] = news;
```

because we need to process a block which has a slug of `featured-post` but data structure is not in a way we want.

File: apps/charterafrica/src/payload/blocks/FeaturedPost.js (hunk 15 above)
Comment (koechkevin) [reply apps/charterafrica/src/payload/blocks/FeaturedPost.js:25]: Got it

File: apps/charterafrica/src/components/RichText/RichText.js [hunk 21]
```diff
@@ -100,7 +100,9 @@ const RichText = React.forwardRef(function RichText(props, ref) {
     return null;
   }
   return (
-    <React.Fragment ref={ref}>{serialize(elements, other)}</React.Fragment>
+    <LineClampedRichTypography ref={ref} {...other}>
+      {serialize(elements, other)}
+    </LineClampedRichTypography>
```
Comment (kilemensi) [inline apps/charterafrica/src/components/RichText/RichText.js:105]: Why this change?

File: apps/charterafrica/src/lib/data/common/index.js [hunk 22]
```diff
@@ -171,48 +171,47 @@ export async function processPageFellowships({ blocks }) {
...
+  });
+
+  const processArticle = (data) => ({
+    ...data,
+    author: data?.authors?.map(({ fullName }) => fullName).join(", ") ?? null,
+    image: data?.coverImage ?? null,
+    date: new Date(data?.publishedOn).toUTCString(),
     link: {
-      href: "/knowledge/news",
+      href: `${breadcrumbs[breadcrumbs.length - 1]?.url}/${data?.id}`,
```
Comment (kilemensi) [inline apps/charterafrica/src/lib/data/common/index.js:186]: 1. To avoid having bogus/undefined `href`, I think lets check if we indeed do have breadcrumbs. If not, `href` should be `null` so that in the component, we don't use `Link`.
2. I think `slugs` make better URLs than `ids`.

File: apps/charterafrica/src/lib/data/common/index.js [hunk 23]
```diff
@@ -171,48 +171,47 @@ export async function processPageFellowships({ blocks }) {
...
     },
   });
-  blocks.push({
+
+  const articles = docs?.map(processArticle);
+  const rawArticle =
+    blocks.find(({ slug }) => slug === "featured-post")?.featuredPost?.value ??
+    null;
+
```
Comment (kilemensi) [inline apps/charterafrica/src/lib/data/common/index.js:194]:
```suggestion
```

File: apps/charterafrica/src/lib/data/common/index.js [hunk 24]
```diff
@@ -171,48 +171,47 @@ export async function processPageFellowships({ blocks }) {
...
+  if (!rawArticle) {
+    return;
+  }
```
Comment (kilemensi) [inline apps/charterafrica/src/lib/data/common/index.js:197]: Yes, if we don't have feature post, we shouldn't show featured post **BUT** we should still show other posts.

File: apps/charterafrica/src/lib/data/common/index.js [hunk 25]
```diff
@@ -171,48 +171,47 @@ export async function processPageFellowships({ blocks }) {
...
+
+  const featuredNewsPost = processArticle(rawArticle);
+
```
Comment (kilemensi) [inline apps/charterafrica/src/lib/data/common/index.js:200]:
```suggestion
```

File: apps/charterafrica/src/lib/data/common/index.js [hunk 26]
```diff
@@ -171,48 +171,47 @@ export async function processPageFellowships({ blocks }) {
...
+  const featuredPost = {
+    category: "News",
+    ...featuredNewsPost,
+    slug: "featured-post",
+  };
+
```
Comment (kilemensi) [inline apps/charterafrica/src/lib/data/common/index.js:206]:
```suggestion
```

File: apps/charterafrica/src/lib/data/common/index.js [hunk 27]
```diff
@@ -171,48 +171,47 @@ export async function processPageFellowships({ blocks }) {
...
-        height: 696,
-        createdAt: "2023-01-26T11:21:14.868Z",
-        updatedAt: "2023-01-26T11:21:14.868Z",
-        url: `/images/knowledge_${(i % 3) + 1}.jpg`,
-      },
-    })),
-  });
+    articles,
+  };
+
```
Comment (kilemensi) [inline apps/charterafrica/src/lib/data/common/index.js:212]:
```suggestion
```

File: apps/charterafrica/src/lib/data/common/index.js [hunk 28]
```diff
@@ -171,48 +171,47 @@ export async function processPageFellowships({ blocks }) {
...
+  blocks[0] = featuredPost;
+  blocks[1] = news;
```
Comment (kilemensi) [inline apps/charterafrica/src/lib/data/common/index.js:214]: 1. featured-post block should already be on page.
2. We can just `push` news block into blocks i.e. it should go last regardless of whether we have featured-post or not.

File: apps/charterafrica/src/payload/blocks/ExternalEmbed.js [hunk 29]
```diff
@@ -0,0 +1,81 @@
+import blockFields from "../fields/blockFields";
+
+const ExternalEmbed = {
+  slug: "embed",
+  fields: [
+    blockFields({
+      name: "embedField",
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/blocks/ExternalEmbed.js:7]: Lets be consistent with naming... others we are calling `---BlockFields`

File: apps/charterafrica/src/payload/blocks/MediaBlock.js [hunk 30]
```diff
@@ -0,0 +1,20 @@
...
+  slug: "mediaBlock",
+  fields: [
+    blockFields({
+      name: "mediaBlockFields",
+      fields: [
+        {
+          name: "media",
+          type: "upload",
+          relationTo: "media",
+          required: true,
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/blocks/MediaBlock.js:13]: Do we allow any other media than image? Can front-end handle anything other than image?

**Review by kilemensi — COMMENTED** [review]
👍🏽

File: apps/charterafrica/src/payload/blocks/MediaBlock.js (hunk 30 above)
Comment (koechkevin) [reply apps/charterafrica/src/payload/blocks/MediaBlock.js:13]: we can add a filter

Comment (kilemensi) [reply apps/charterafrica/src/payload/blocks/MediaBlock.js:13]: Cool... lets also rename it to `image` if we're only supporting images.

File: apps/charterafrica/src/payload/collections/Tag.js [hunk 31]
```diff
@@ -0,0 +1,21 @@
+const Tag = {
+  slug: "tag",
+  admin: {
+    useAsTitle: "tagName",
+  },
+  fields: [
+    {
+      name: "tagName",
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/collections/Tag.js:8]:
```suggestion
      name: "name",
```

We know it's a tag already.

File: apps/charterafrica/src/payload/collections/Tag.js [hunk 32]
```diff
@@ -0,0 +1,21 @@
...
+      label: {
+        en: "Tag Name",
+        fr: "Nom de la balise",
+        pt: "Nome da etiqueta",
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/collections/Tag.js:12]:
```suggestion
        en: "Name",
        fr: "Nom",
        pt: "Nome",
```

**Review by kilemensi — COMMENTED** [review]
👍🏽

**Comment by kilemensi** [comment]
I'm also getting hydration errors on this branch @koechkevin :
Screenshot 2023-02-27 at 15-21-06 charter AFRICA

**Comment by kilemensi** [comment]
@koechkevin ?

**Comment by koechkevin** [comment]
> I'm also getting hydration errors on this branch @koechkevin : Screenshot 2023-02-27 at 15-21-06 charter AFRICA

@kelvinkipruto could this be similar cause to what you are facing?

**Comment by koechkevin** [comment]
> > I'm also getting hydration errors on this branch @koechkevin : Screenshot 2023-02-27 at 15-21-06 charter AFRICA
> 
> @kelvinkipruto could this be similar cause to what you are facing?

also @kilemensi on which path did you get this error?

File: apps/charterafrica/src/components/RichText/RichText.js (hunk 21 above)
Comment (koechkevin) [reply apps/charterafrica/src/components/RichText/RichText.js:105]: If you paste a long content with paragraphs and have lineClamp property, I'd end up with something like this.

image

**Comment by kilemensi** [comment]
> > > I'm also getting hydration errors on this branch @koechkevin : Screenshot 2023-02-27 at 15-21-06 charter AFRICA
> > 
> > 
> > @kelvinkipruto could this be similar cause to what you are facing?
> 
> also @kilemensi on which path did you get this error?

This was on homepage @koechkevin ... I haven't tried other pages though.

**Comment by kelvinkipruto** [comment]
> This was on homepage @koechkevin ... I haven't tried other pages though.

@kilemensi @koechkevin I'm also facing the issue on my #343. In my case, I have narrowed it down to this [code](https://github.com/CodeForAfrica/ui/blob/07538798bbad8078d6ebae8515f9ba13855e77a3/apps/charterafrica/src/components/OpportunityCard/OpportunityCard.js#L89-L99), which when removed, solves it. Still trying to fix it.

File: apps/charterafrica/src/lib/data/common/index.js [hunk 33]
```diff
@@ -171,48 +171,42 @@ export async function processPageFellowships({ blocks }) {
...
+  const { blocks, breadcrumbs = [] } = page;
+  const { docs } = await api.getCollection("news", {
+    where: { _status: { equals: "published" } },
+  });
+
+  const processArticle = (data) => ({
+    ...data,
+    author: data?.authors?.map(({ fullName }) => fullName).join(", ") ?? null,
+    image: data?.coverImage ?? null,
+    date: new Date(data?.publishedOn).toUTCString(),
```
Comment (kelvinkipruto) [inline apps/charterafrica/src/lib/data/common/index.js:184]: Wouldn't be better to use ```toLocaleString()```

File: apps/charterafrica/src/payload/collections/Tag.js [hunk 34]
```diff
@@ -0,0 +1,21 @@
+const Tag = {
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/collections/Tag.js:1]:
```suggestion
const Tags = {
```

File: apps/charterafrica/src/payload/collections/Tag.js [hunk 35]
```diff
@@ -0,0 +1,21 @@
...
+  slug: "tag",
+  admin: {
+    useAsTitle: "tagName",
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/collections/Tag.js:4]:
```suggestion
    useAsTitle: "name",
```

File: apps/charterafrica/src/payload/collections/Tag.js [hunk 36]
```diff
@@ -0,0 +1,21 @@
...
+  },
+  fields: [
+    {
+      name: "name",
+      label: {
+        en: "Tag Name",
+        fr: "Nom de la balise",
+        pt: "Nome da etiqueta",
+      },
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/collections/Tag.js:13]: Add `slug` field as well.

File: apps/charterafrica/src/components/RichText/RichText.js [hunk 37]
```diff
@@ -31,7 +27,10 @@ const serialize = (children, props) =>
     if (!node) {
       return null;
     }
-
+    const DEFAULT_PROPS = {
+      html: false,
+      ...props,
+    };
```
Comment (kilemensi) [inline apps/charterafrica/src/components/RichText/RichText.js:33]: Nope...

1. `DEFAULT_PROPS` is a `const` hence caps.
2. By spreading both `DEFAULT_PROPS` followed by `props` on component, we achieve the same thing without having to create a throwaway object.

File: apps/charterafrica/src/components/RichText/RichText.js [hunk 38]
```diff
@@ -99,9 +98,7 @@ const RichText = React.forwardRef(function RichText(props, ref) {
   if (!elements?.length) {
     return null;
   }
-  return (
-    <React.Fragment ref={ref}>{serialize(elements, other)}</React.Fragment>
-  );
+  return <Fragment ref={ref}>{serialize(elements, other)}</Fragment>;
```
Comment (kilemensi) [inline apps/charterafrica/src/components/RichText/RichText.js:101]: `Box` that you have on `FeaturedPost` should replace Fragment here. We'll then use
```jsx
    sx={{
      "& h1": ...
    }}
```

to style content as needed.

File: apps/charterafrica/src/lib/data/common/index.js [hunk 39]
```diff
@@ -171,48 +171,42 @@ export async function processPageFellowships({ blocks }) {
...
+  const { blocks, breadcrumbs = [] } = page;
+  const { docs } = await api.getCollection("news", {
+    where: { _status: { equals: "published" } },
+  });
+
+  const processArticle = (data) => ({
+    ...data,
+    author: data?.authors?.map(({ fullName }) => fullName).join(", ") ?? null,
+    image: data?.coverImage ?? null,
+    date: new Date(data?.publishedOn).toLocaleString(),
```
Comment (kilemensi) [inline apps/charterafrica/src/lib/data/common/index.js:184]: Should we pass the current `locale` to `toLocaleString`?

File: apps/charterafrica/src/payload/blocks/ExternalEmbed.js [hunk 40]
```diff
@@ -0,0 +1,81 @@
+import blockFields from "../fields/blockFields";
+
+const ExternalEmbed = {
+  slug: "embed",
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/blocks/ExternalEmbed.js:4]: This should match ExternalEmbed with external-embed or Embed with embed.

File: apps/charterafrica/src/payload/blocks/MediaBlock.js [hunk 41]
```diff
@@ -0,0 +1,23 @@
+import blockFields from "../fields/blockFields";
+
+const MediaBlock = {
+  slug: "mediaBlock",
+  fields: [
+    blockFields({
+      name: "mediaBlockFields",
+      fields: [
+        {
+          name: "media",
```
Comment (kilemensi) [inline apps/charterafrica/src/payload/blocks/MediaBlock.js:10]:
```suggestion
          name: "image",
```
This way, FE will know this block is just for image (and not audio or video or PDF file, etc.)

**Review by kilemensi — COMMENTED** [review]
👍🏽

File: apps/charterafrica/src/components/FeaturedPostCard/FeaturedPostCard.js [hunk 42]
```diff
@@ -96,24 +97,33 @@ const FeaturedPostCard = React.forwardRef(function FeaturedPostCard(
...
               })}
             </Typography>
-
-            <LineClampedRichTypography
+            {/* Line height is constant across all screens at 24 px * 6 lines. Has to be wrapped in another div to overcome */}
+            <RichText
+              sx={{
+                maxHeight: 24 * 6,
+                overflow: "hidden",
+                "& :is(h1, h2, h3, h4, h5, h6)": {
```
Comment (kilemensi) [inline apps/charterafrica/src/components/FeaturedPostCard/FeaturedPostCard.js:105]: Not sure about this.

File: apps/charterafrica/src/components/FeaturedPostCard/FeaturedPostCard.js [hunk 43]
```diff
@@ -96,24 +97,33 @@ const FeaturedPostCard = React.forwardRef(function FeaturedPostCard(
                 day: "numeric",
               })}
             </Typography>
-
-            <LineClampedRichTypography
+            {/* Line height is constant across all screens at 24 px * 6 lines. Has to be wrapped in another div to overcome */}
```
Comment (kilemensi) [inline apps/charterafrica/src/components/FeaturedPostCard/FeaturedPostCard.js:100]: We don't need this since `FeaturedPost` only appears in desktop so it should go without saying.

**Review by kilemensi — APPROVED** [review]
🚀
